controllers/nstemplatetierrevisioncleanup/nstemplatetier_revision_cleanup_controller_test.go

```diff
@@ -0,0 +1,276 @@
+		ttr := createttr(*nsTemplateTier, (nsTemplateTier.Spec.ClusterResources.TemplateRef + "-ttrcr"), metav1.NewTime(time.Now().Add(-time.Minute)))
+		s := createSpace(nsTemplateTier)
+		r, req, cl := prepareReconcile(t, ttr.Name, ttr, s, nsTemplateTier)
+		//when
+		res, err := r.Reconcile(context.TODO(), req)
+		//then
+		require.NoError(t, err)
+		require.Equal(t, controllerruntime.Result{}, res)
+		tiertemplaterevision.AssertThatTTRs(t, cl, nsTemplateTier.GetNamespace()).DoNotExist()
+	})
+
+	t.Run("Failure", func(t *testing.T) {
+		nsTemplateTier := tiertest.Base1nsTier(t, tiertest.CurrentBase1nsTemplates, tiertest.WithStatusRevisions())
+		ttr := createttr(*nsTemplateTier, (nsTemplateTier.Spec.ClusterResources.TemplateRef + "-ttrcr"), metav1.NewTime(time.Now().Add(-time.Minute)))
+		s := createSpace(nsTemplateTier)
+		t.Run("the creation timestamp is less than 30 sec", func(t *testing.T) {
+			// given
+			ttr := createttr(*nsTemplateTier, (nsTemplateTier.Spec.ClusterResources.TemplateRef + "-ttr"), metav1.NewTime(time.Now().Add(-29*time.Second)))
+			r, req, cl := prepareReconcile(t, ttr.Name, ttr, s, nsTemplateTier)
+
+			// when
+			res, err := r.Reconcile(context.TODO(), req)
+
+			// then
+			require.NoError(t, err)
+			require.LessOrEqual(t, res.RequeueAfter, time.Second)
+			tiertemplaterevision.AssertThatTTRs(t, cl, nsTemplateTier.GetNamespace()).ExistFor(nsTemplateTier.Name)
+
+		})
+		t.Run("ttr is still being referenced in status.revisions", func(t *testing.T) {
+			// given
+			ttr := createttr(*nsTemplateTier, (nsTemplateTier.Spec.ClusterResources.TemplateRef + "-ttr"), metav1.NewTime(time.Now().Add(-time.Minute)))
+			r, req, cl := prepareReconcile(t, ttr.Name, ttr, s, nsTemplateTier)
+
+			// when
+			res, err := r.Reconcile(context.TODO(), req)
+
+			// then
+			require.NoError(t, err)
+			require.Equal(t, controllerruntime.Result{}, res)
+			tiertemplaterevision.AssertThatTTRs(t, cl, nsTemplateTier.GetNamespace()).ExistFor(nsTemplateTier.Name)
+
+		})
+
+		t.Run("spaces are still being updated", func(t *testing.T) {
+			// given
+			nsTemplateTier.Status.Revisions = map[string]string{
+				"base1ns-code-123456new":             "base1ns-code-123456new-ttr",
+				"base1ns-clusterresources-123456new": "base1ns-clusterresources-123456new-ttr",
+			}
+			ttr := createttr(*nsTemplateTier, (nsTemplateTier.Spec.ClusterResources.TemplateRef + "-ttrcr"), metav1.NewTime(time.Now().Add(-time.Minute)))
+			r, req, cl := prepareReconcile(t, ttr.Name, ttr, s, nsTemplateTier)
+
+			// when
+			res, err := r.Reconcile(context.TODO(), req)
+
+			// then
+			require.NoError(t, err)
+			require.Equal(t, controllerruntime.Result{}, res)
+			tiertemplaterevision.AssertThatTTRs(t, cl, nsTemplateTier.GetNamespace()).ExistFor(nsTemplateTier.Name)
+		})
+
+		t.Run("Error while deleting the TTR", func(t *testing.T) {
+			// given
+			nsTemplateTier := tiertest.Base1nsTier(t, tiertest.CurrentBase1nsTemplates, tiertest.WithStatusRevisions())
```

Comment:
we could use the one from the parent test here
```suggestion
```

Reply:
Done PTAL